media_driver/linux/common/vp/ddi/media_libva_vp.c

```diff
@@ -836,8 +836,9 @@ VAStatus VpUpdateProcHdrState(
                 pVpHalSurf->pHDRParams->white_point_y = pHDR10MetaData->white_point_y;
                 VP_DDI_NORMALMESSAGE("pHDR10MetaData white_point_x %d, white_point_y %d.", pHDR10MetaData->white_point_x, pHDR10MetaData->white_point_y);
 
-                pVpHalSurf->pHDRParams->max_display_mastering_luminance = pHDR10MetaData->max_display_mastering_luminance;
-                pVpHalSurf->pHDRParams->min_display_mastering_luminance = pHDR10MetaData->min_display_mastering_luminance;
+                // From VAAPI defintion which is following video spec, max/min_display_mastering_luminance are in units of 0.0001 candelas per square metre.
+                pVpHalSurf->pHDRParams->max_display_mastering_luminance = (uint16_t)(pHDR10MetaData->max_display_mastering_luminance/10000);
+                pVpHalSurf->pHDRParams->min_display_mastering_luminance = (uint16_t)(pHDR10MetaData->min_display_mastering_luminance/10000);
```

Comment:
If user sets the uint_32_t vaapi values > 655350000 then there will be an overflow.  Perhaps we should validate the user data here.

Reply:
Updated.